pkg/relay/ethereum/chain.go

```diff
@@ -474,6 +474,42 @@ func (c *Chain) QueryUnfinalizedRelayAcknowledgements(ctx core.QueryContext, cou
 	return packets, nil
 }
 
+// QueryChannelUpgrade returns the channel upgrade associated with a channelID
+func (c *Chain) QueryChannelUpgrade(ctx core.QueryContext) (*chantypes.QueryUpgradeResponse, error) {
+	if upg, found, err := c.ibcHandler.GetChannelUpgrade(c.callOptsFromQueryContext(ctx), c.pathEnd.PortID, c.pathEnd.ChannelID); err != nil {
+		return nil, err
+	} else if !found {
+		return nil, nil
+	} else {
+		return chantypes.NewQueryUpgradeResponse(upgradeToPB(upg), nil, ctx.Height().(clienttypes.Height)), nil
+	}
+}
+
+// QueryChannelUpgradeError returns the channel upgrade error receipt associated with a channelID at the height of `ctx`.
+// WARN: This error receipt may not be used to cancel upgrade in FLUSHCOMPLETE state because of upgrade sequence mismatch.
+func (c *Chain) QueryChannelUpgradeError(ctx core.QueryContext) (*chantypes.QueryUpgradeErrorResponse, error) {
+	if ev, err := c.findWriteErrorReceipt(ctx); err != nil {
+		return nil, err
+	} else if ev == nil {
+		return nil, nil
+	} else {
+		return &chantypes.QueryUpgradeErrorResponse{
+			ErrorReceipt: chantypes.ErrorReceipt{
+				Sequence: ev.UpgradeSequence,
+				Message:  ev.Message,
+			},
+			ProofHeight: clienttypes.NewHeight(0, ev.Raw.BlockNumber),
```

Comment:
Do we need to set this proof height?

Reply:
No! Thank you I will delete this line.

Reply:
@bluele done: https://github.com/datachainlab/ethereum-ibc-relay-chain/pull/50/commits/91f370e31062b4e6f446dc059a412eb96b272da8